### ISSUE TITLE
`CSVLogger` log first (global_step = 0) iteration

### ISSUE THREAD

**Issue by skanti** [PyTorchLightning/pytorch-lightning#8384]
## 🚀 Feature
See this [issue](https://github.com/PyTorchLightning/pytorch-lightning/issues/8098). The current `CSVLogger` ignores the very first iteration and evaluation on the sanity validation set.

### Motivation

The solution of using `log_every_n_steps=1` is not always feasible. E.g. Performance hit when your logging directory is on a (slow) remote server.

Seems like `flush_logs_every_n_steps` is only implemented for `tensorboard`(?). To make it work for `CSVLogger` is okay too - would prefer to generally log the first iterations.

**Comment by ananyahjha93** [comment]
@skanti hey, so I think like @justusschock mentioned, these two parameters work according to the intended behavior. Now, it's a question of whether we want logging on the very first step as an exception to this rule. 

Tagging @tchaton @Borda @justusschock  for their opinion on this.

**Comment by skanti** [comment]
Objectively looking at it:
Hard to think of benefits of discarding information/data. I mean there is not even an option to not discard the first iteration.
One use case particularly interesting for me is to observe the absolute drop of my train/val losses (from the first iteration).

Also:
How are `num_sanity_val_steps` logged for `log_every_n_steps > 1`? Seems to log only for `log_every_n_steps = 1`
